### PR TITLE
Features/v5/UUID

### PR DESCRIPTION
**Pull request by Sauilitired**
Rewrite UUID and player management.

### CODE REVIEW

**Comment by darbyjack** [comment]
Looks good.

File: Bukkit/src/main/java/com/plotsquared/bukkit/listener/PaperListener.java [hunk 1]
```diff
@@ -306,4 +316,40 @@ public void onPlayerNaturallySpawnCreaturesEvent(PlayerNaturallySpawnCreaturesEv
             event.setCancelled(true);
         }
     }
+
+    @EventHandler public void onAsyncTabCompletion(final AsyncTabCompleteEvent event) {
+        String buffer = event.getBuffer();
```
Comment (Sauilitired) [inline Bukkit/src/main/java/com/plotsquared/bukkit/listener/PaperListener.java:321]: Needs a config option

File: Bukkit/src/main/java/com/plotsquared/bukkit/player/BukkitPlayer.java [hunk 2]
```diff
@@ -97,10 +96,7 @@ public BukkitPlayer(@NotNull final Player player, final boolean offline) {
     }
 
     @NotNull @Override public UUID getUUID() {
-        if (this.uuid == null) {
```
Comment (Sauilitired) [inline Bukkit/src/main/java/com/plotsquared/bukkit/player/BukkitPlayer.java:100]: Re-implement this

File: Bukkit/src/main/java/com/plotsquared/bukkit/util/uuid/OfflineUUIDWrapper.java [hunk 3]
```diff
@@ -1,131 +0,0 @@
...
-import org.bukkit.entity.Player;
-import org.jetbrains.annotations.NotNull;
-
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.UUID;
-
-public class OfflineUUIDWrapper extends UUIDWrapper {
```
Comment (Sauilitired) [inline Bukkit/src/main/java/com/plotsquared/bukkit/util/uuid/OfflineUUIDWrapper.java:49]: Bring back.

File: Core/src/main/java/com/plotsquared/core/configuration/Settings.java [hunk 4]
```diff
@@ -239,6 +239,16 @@ public static boolean convertLegacy(File file) {
         @Comment("Use a database to store UUID/name info") public static boolean
             USE_SQLUUIDHANDLER = false;
         @Ignore public static boolean NATIVE_UUID_PROVIDER = false;
```
Comment (Sauilitired) [inline Core/src/main/java/com/plotsquared/core/configuration/Settings.java:241]: Remove old unused fields.

**Comment by Sauilitired** [comment]
Commands have been tested and appear to work like intended.

**Pull request merged**